torch/_dynamo/utils.py

```diff
@@ -750,6 +750,10 @@ def torch_clone(x):
             # Access data_ptr() for a xla tensor will cause crash
             return torch_clone(x)
 
+        if x.layout is not torch.strided:
+            # Anything without a stride is directly cloned.
```

Comment:
I wonder if it will be relevant to preserve `values` `and `indices` strides as well?

Reply:
is there a deep clone for that?

Reply:
`sparse.clone()` is deep clone. Iff there is a need to preserve the strides of indices/values, the corresponding implementation bit in `clone_input` function would be something like:
```python
if x.layout is torch.sparse_csr:
   compressed_indices = torch_clone(x.crow_indices())
   plain_indices = torch_clone(x.col_indices())
   values = torch_clone(x.values())
   return torch.sparse_compressed_tensor(compressed_indices, plain_indices, values, x.shape, layout=x.layout)
elif ...:
    ...
```

I'd leave this discussion open until its importance is proved in one or other way.

Reply:
Added the TODO, so closing the discussion just for clarity